### DISCUSSION TITLE
Dagger should enable network resilience everywhere

### DISCUSSION THREAD

**Discussion by verdverm** [dagger/dagger#12052] [Ideas]
My current build system often fails when a remote network / service behaves badly. 502's, host hangs up, services being down... Today, I have to add custom logic at all of these points as we discover them, or we move to replicating these remotes internally.

- `git clone`
- `go mod download`
- `docker pull`
- `apt update`
- the list goes on and on...

It would be great if there was retry / expo decay options across the building blocks for these situations. Detecting which are network errors vs which are not could be difficult. Maybe some sort of retry / decay wrapper that could support rerunning an `#up` after checks defined in CUE (like string processing). Either way, having a consistent and reusable solution would be Grrrreat!

**Comment by aluzzardi** [comment]
Also requested by @evalsocket & others in https://github.com/dagger/dagger/discussions/2052#discussioncomment-2536712

**Comment by sipsma** [comment]
Just as another example of this, @jpadams hit a remote cache import error (`failed to compute cache key: failed to copy: invalid status response 503 Egress is over the account limit`) due to a 503 from GHA that failed the entire run. In general Buildkit tries to treat these errors as cache misses, but due to the way this failed Buildkit didn't. Have an issue open there now, though it's actually a pretty hard problem: https://github.com/moby/buildkit/issues/2836
